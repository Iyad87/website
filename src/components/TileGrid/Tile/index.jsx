import React from 'react';
import PropTypes from 'prop-types';
import { Container, Content, ContentTitle, ContentBody } from './styled';

const Tile = ({ colour }) => (
  <Container colour={colour}>
    <Content>
      <ContentTitle>hello</ContentTitle>
      <ContentBody>hello, world! i represent some tiled content</ContentBody>
    </Content>
  </Container>
);

Tile.propTypes = {
  colour: PropTypes.oneOf(['normal', 'dark', 'darker']),
};

Tile.defaultProps = {
  colour: 'normal',
};

export default Tile;
