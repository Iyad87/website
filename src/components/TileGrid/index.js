import styled from 'styled-components';
import Tile from './Tile';

const TileGrid = styled.ul`
  display: flex;
  flex-wrap: wrap;
  padding: 0;
`;

export default TileGrid;

export { Tile };
