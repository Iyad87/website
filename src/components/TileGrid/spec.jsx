import React from 'react';
import { shallow } from 'enzyme';
import TileGrid from './index';

test('components/TileGrid snapshot', () => {
  const wrapper = shallow(<TileGrid />);

  expect(wrapper).toMatchSnapshot();
});
