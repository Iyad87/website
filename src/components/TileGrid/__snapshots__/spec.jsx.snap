// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/TileGrid snapshot 1`] = `
<ul
  className="sc-ifAKCX ZzcRA"
/>
`;
